src/app/features/conditions/edit-conditions/conditions.component.ts

```diff
@@ -44,6 +47,10 @@ export class ConditionsComponent extends SingleRowEditorComponent<Conditions> {
     return this.editorService.form.controls.ConditionTypeOrReference.value;
   }
 
+  get showQuestState(): boolean {
+    return this.conditionType === CONDITION_TYPES.CONDITION_QUESTSTATE;
+  }
```

Comment:
@Kitzunu  3) small method that is telling us when to show the quest state flag selector (according to the value of condition type)